### PR TITLE
Add aws/kms + minio/sio symmetric client side encryption.

### PR DESCRIPTION
**Pull request by bn0ir**
Add aws/kms + minio/sio symmetric client side encryption.

`WALG_CSE_KMS_ID` env var specifies KMS key id.
`WALG_CSE_KMS_REGION` env var specifies KMS key region.

Encryption:

- Generate unique symmetric key for every backup;
- Encrypt key with AWS KMS;
- Encrypt backup with symmetric key (minio/sio DARE 32 bytes key);
- Add encrypted symmetric key as backup file prefix (184 bytes);

Decryption:

- Read encrypted symmetric key from file prefix (184 bytes);
- Decrypt symmetric key with AWS KMS;
- Decrypt file with symmetric key (minio/sio DARE 32 bytes key);

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,6 +116,13 @@ To enable S3 server-side encryption, set to the algorithm to use when storing th
 
 If using S3 server-side encryption with `aws:kms`, the KMS Key ID to use for object encryption.
 
+* `WALG_CSE_KMS_ID`
```
Comment (x4m) [inline README.md]: These settings, they are S3-only, aren't they? If they are, I think we should use WALG_S3_CSE_...

File: internal/crypto/awskms/crypter.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+	return ioextensions.NewOnCloseFlusher(encryptedWriter, bufferedWriter), nil
+}
+
+// Decrypt creates decrypted reader from ordinary reader
+func (crypter *Crypter) Decrypt(reader io.Reader) (io.Reader, error) {
+	encryptedSymmetricKey := make([]byte, crypter.SymmetricKey.GetEncryptedKeyLen())
+	_, err := reader.Read(encryptedSymmetricKey)
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Can't read encryption key from s3: %v", err)
```
Comment (x4m) [inline internal/crypto/awskms/crypter.go]: I think we should better wrap error with our description...

File: internal/crypto/awskms/crypter.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		return reader, err
+	}
+
+	crypter.SymmetricKey.SetEncryptedKey(encryptedSymmetricKey)
+
+	err = crypter.SymmetricKey.Decrypt()
+	if err != nil {
+		tracelog.ErrorLogger.Printf("Can't decrypt symmetric key: %v", err)
```
Comment (x4m) [inline internal/crypto/awskms/crypter.go]: Is this fatal? Looks like we continue anyway....

File: README.md (hunk 1 above)
Comment (bn0ir) [reply README.md]: In most cases you want to use it with s3, but they can be used with any storage backend (you can use AWS KMS encryption and GCP storage)

File: internal/crypto/awskms/crypter.go (hunk 3 above)
Comment (bn0ir) [reply internal/crypto/awskms/crypter.go]: In most cases this error will be fatal, minio/sio will fail with error

File: internal/crypto/awskms/crypter.go (hunk 2 above)
Comment (bn0ir) [reply internal/crypto/awskms/crypter.go]: Fix error message in https://github.com/wal-g/wal-g/pull/315/commits/360633709839d3e0adda77fc760237d8fe924fb4

File: internal/crypto/awskms/crypter.go (hunk 3 above)
Comment (bn0ir) [reply internal/crypto/awskms/crypter.go]: Fix in https://github.com/wal-g/wal-g/pull/315/commits/f36b26391cf3aa6c3c467411da9d33af699641a7

**Comment by x4m** [comment]
Thanks! Pushed

**Pull request merged**